lib/utils/get-module-path-for.js

```diff
@@ -18,12 +18,32 @@ for (let packagePath of packagePaths) {
   let packageDir = path.dirname(path.resolve(cwd, packagePath));
 
   if (pkg.keywords && pkg.keywords.includes('ember-addon')) {
-    ADDON_PATHS[packageDir] = pkg.name;
+    // build addon instance
+    ADDON_PATHS[packageDir] = getAddonPackageName(packagePath);
   } else if (isEmberCliProject(pkg)) {
     APP_PATHS[packageDir] = pkg.name;
   }
 }
 
+/**
+ * takes a package path and returns the runtime name of the addon.
+ *
+ * @param {String} packagePath the path on disk (from current working directory)
+ * @returns {String} The runtime name of the addon
+ */
+function getAddonPackageName(packagePath) {
+  const indexPath = path.resolve(path.dirname(packagePath), 'index.js');
+
+  if (fs.existsSync(indexPath)) {
+    const { name: packageName, moduleName } = require(indexPath);
+    // this is bad, fix later
+    return moduleName && typeof moduleName === 'function' ? moduleName() : packageName;
+  }
+
+  const { name: packageName } = require(packagePath);
+  return packageName;
+}
```

Comment:
I think something like this will accommodate my concerns:

```suggestion
function getAddonPackageName(packagePath) {
  const pkg = require(packagePath);
  const entryPoint = pkg['ember-addon'] && pkg['ember-addon'].main
    ? pkg['ember-addon'].main
    : 'index.js';

  let moduleName = pkg.name;
  try {
    let entryModule = require(path.join(packagePath, entryPoint));
    if (typeof entryModule.moduleName === 'function') {
      moduleName = entryModule.moduleName();
    }
  } catch {
    // do nothing, this falls back to using package name
  }
  
  return moduleName;
}
```


What do you think @hmajoros?

Reply:
sounds good to me 👍